database/migrations/platform/2024_03_17_000000_update_subscription_table.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+return new class() extends Migration {
+    public function up(): void
+    {
+        // [1] Create `id` column as the new primary key.
+        // SQLite doesn't let us change a primary key after the initial migration.
+        /** @see 2012_10_03_133633_create_base_tables.php */
+        Schema::table('Subscription', function (Blueprint $table) {
+            if (DB::connection()->getDriverName() !== 'sqlite') {
+                $table->dropPrimary(['SubjectType', 'SubjectID', 'UserID']);
+            }
+        });
+        Schema::table('Subscription', function (Blueprint $table) {
+            if (DB::connection()->getDriverName() !== 'sqlite') {
+                $table->bigIncrements('id')->first();
+            }
+        });
+
+        // [2] Rename columns to align with Laravel conventions.
+        Schema::table('Subscription', function (Blueprint $table) {
+            $table->renameColumn('SubjectType', 'subject_type');
+        });
+        Schema::table('Subscription', function (Blueprint $table) {
+            $table->renameColumn('SubjectID', 'subject_id');
+        });
+        Schema::table('Subscription', function (Blueprint $table) {
+            $table->renameColumn('UserID', 'user_id');
+        });
+        Schema::table('Subscription', function (Blueprint $table) {
+            $table->renameColumn('State', 'state');
+        });
+        Schema::table('Subscription', function (Blueprint $table) {
+            $table->renameColumn('Created', 'created_at');
+        });
+        Schema::table('Subscription', function (Blueprint $table) {
+            $table->renameColumn('Updated', 'updated_at');
+        });
+
+        // [3] Change the datatype of `user_id` to match `UserAccounts.ID`.
+        Schema::table('Subscription', function (Blueprint $table) {
+            $table->unsignedBigInteger('user_id')->change();
+        });
+
+        // [4] Enforce a unique constraint on type, subject_id, and user_id combos.
+        Schema::table('Subscription', function (Blueprint $table) {
+            $table->unique(['type', 'subject_id', 'user_id']);
```

Comment:
This failed for me.
```
SQLSTATE[42000]: Syntax error or access violation: 1072 Key column 'type' doesn't exist in table (Connection:
mysql, SQL: alter table `Subscription` add unique `subscription_type_subject_id_user_id_unique`(`type`,
`subject_id`, `user_id`))
```

I think it should be `subject_type`.

Reply:
Yep, sorry about that. Fixed in https://github.com/RetroAchievements/RAWeb/pull/2298/commits/17c28f727c2c9ec8d14c51d051e08c5f0843f4ad.